packages/http-server/src/__tests__/server.oas.spec.ts

```diff
@@ -228,6 +295,67 @@ describe.each([['petstore.oas2.json'], ['petstore.oas3.json']])('server %s', fil
     }
   });
 
+  describe('server validation: given __server query param', () => {
+    it('when the server is not valid then return error', async () => {
+      const response = await server.fastify.inject({
+        method: 'GET',
+        url: '/pets/10?__server=https://google.com',
+      });
+
+      expect(response.statusCode).toBe(404);
+      expect(response.payload).toEqual(
+        '{"type":"https://stoplight.io/prism/errors#NO_SERVER_MATCHED_ERROR","title":"Route not resolved, no server matched.","status":404,"detail":"The base url https://google.com hasn\'t been matched with any of the provided servers"}',
+      );
+    });
+
+    it('when the server is valid then return 200', async () => {
+      const response = await server.fastify.inject({
+        method: 'GET',
+        url: '/pets/10?__server=https://petstore.swagger.io/v2',
+      });
+
+      expect(response.statusCode).toBe(200);
+    });
+
+    // oas2 does not support overriding servers
+    if (file === 'petstore.oas3.json') {
```

Comment:
This is ok for now, but maybe we could have oas3 specific tests instead of branching